docs/operators/transform.md

```diff
@@ -0,0 +1,121 @@
+    -   Default Value = {java.lang.Math.*, org.apache.commons.lang3.StringUtils.*, org.apache.commons.lang3.StringEscapeUtils.*, org.apache.commons.lang3.time.DurationFormatUtils.*, org.apache.commons.lang3.time.DateFormatUtils.*}
+
+-   ***copyMatchingFields*** -   boolean
+    -   Specifies whether the matching fields between input and output should be copied as is or not.
+    -   Default Value = true.
+    
+#### Ports
+----------
+
+-   ***input*** -   Port for input tuples.
+-   ***output***    -   Port for transformed output tuples.
+
+#### Attributes
+---------------
+-   ***Input port Attribute - input.TUPLE\_CLASS*** - Fully qualified class name.
+    -   Mandatory attribute
+    -   Type of input tuple.
+
+-   ***Output port Attribute - output.TUPLE\_CLASS*** - Fully qualified class name.
+    -   Mandatory attribute
+    -   Type of output tuple.
+    
+#### Application Example
+------------------------
+
+Please refer [Example] (https://github.com/DataTorrent/examples/tree/master/tutorials/transform) for transform sample application.
+
+### Partitioning of TransformOperator
+Being stateless operator, TransformOperator will ensure built-in partitioners present in Malhar library can be directly by setting few properties as follows:
+
+#### Stateless partitioning of TransformOperator
+Stateless partitioning will ensure that TransformOperator will will be partitioned right at the starting of the application and will remain partitioned throughout the lifetime of the DAG.
+TransformOperator can be stateless partitioned by adding following lines to properties.xml:
+
+```xml
+  <property>
+    <name>dt.operator.{OperatorName}.attr.PARTITIONER</name>
+    <value>com.datatorrent.common.partitioner.StatelessPartitioner:3</value>
+  </property>
+```
+
+where {OperatorName} is the name of the TransformOperator operator.
+Above lines will partition TransformOperator statically 3 times. Above value can be changed accordingly to change the number of static partitions.
+
+#### Dynamic Partitioning of TransformOperator
+Dynamic partitioning is a feature of Apex platform which changes the partition of the operator based on certain condition.
+TransformOperator can be dynamically partitioned using the below two partitioners:
+
+##### Throughput based
+Following code can be added to populateDAG method of application to dynamically partitioning TransformOperator:
+```java
+    StatelessThroughputBasedPartitioner<TransformOperator> partitioner = new StatelessThroughputBasedPartitioner<>();
+    partitioner.setCooldownMillis(conf.getLong(COOL_DOWN_MILLIS, 10000));
+    partitioner.setMaximumEvents(conf.getLong(MAX_THROUGHPUT, 30000));
+    partitioner.setMinimumEvents(conf.getLong(MIN_THROUGHPUT, 10000));
+    dag.setAttribute(transform, OperatorContext.STATS_LISTENERS, Arrays.asList(new StatsListener[]{partitioner}));
+    dag.setAttribute(transform, OperatorContext.PARTITIONER, partitioner);
+```
```

Comment:
Setting the properties to the operator attribute. I think this can't be done from properties file.